submit_and_compare/submit_and_compare.py

```diff
@@ -289,17 +289,16 @@ def student_submit(self, submissions, suffix=''):
         Save student answer
         """
         # when max_attempts == 0, the user can make unlimited attempts
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
+        success = False
+	if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
             LOG.error(
                 'User has already exceeded the maximum '
                 'number of allowed attempts',
             )
-            result = {
-                'success': False,
-                'problem_progress': self._get_problem_progress(),
-                'submit_class': self._get_submit_class(),
-                'used_attempts_feedback': self._get_used_attempts_feedback(),
-            }
+	elif self.is_past_due():
+	    LOG.error(
```

Comment:
Decide whether or not we need logging here and if so,
which level of verbosity is appropriate [1].

[1] https://docs.python.org/2/library/logging.html